admin/server/reports.go

```diff
@@ -269,12 +270,21 @@ func (s *Server) UnsubscribeReport(ctx context.Context, req *adminv1.Unsubscribe
 		return nil, status.Error(codes.FailedPrecondition, "can't edit report because it was not created from the UI")
 	}
 
-	if claims.OwnerType() != auth.OwnerTypeUser {
+	if claims.OwnerType() != auth.OwnerTypeUser && claims.OwnerType() != auth.OwnerTypeMagicAuthToken {
 		return nil, status.Error(codes.PermissionDenied, "only users can unsubscribe from reports")
 	}
-	user, err := s.admin.DB.FindUser(ctx, claims.OwnerID())
-	if err != nil {
-		return nil, status.Error(codes.Internal, err.Error())
+
+	var userEmail string
+	if claims.OwnerType() == auth.OwnerTypeUser {
+		user, err := s.admin.DB.FindUser(ctx, claims.OwnerID())
+		if err != nil {
+			return nil, status.Error(codes.Internal, err.Error())
+		}
+		userEmail = user.Email
+	} else if req.Email == nil {
+		return nil, status.Error(codes.InvalidArgument, "email is required for magic auth token")
+	} else {
+		userEmail = *req.Email
```

Comment:
This doesn't check that the email belongs to the current magic auth token. I think it would be better to do a DB lookup for the current magic auth token's email (can use `claims.OwnerID()` for the magic auth token ID).